uint8 SUCCEEDED = 0
uint8 SKIPPED = 1
uint8 FAILED = 2
uint8 status